### PR TITLE
feat: Add additional replacements necessary for reprocessing

### PR DESCRIPTION
**Pull request by untitaker**
This adds additional message types such that Sentry's reprocessing-v2 can reuse event IDs for the reprocessed events. See `*_insert` testcase which goes through the exact flow.

### CODE REVIEW

File: snuba/datasets/errors_replacer.py [hunk 1]
```diff
@@ -236,21 +230,83 @@ def process_delete_groups(
...
 
 
+def process_exclude_groups(message: Mapping[str, Any]) -> Optional[Replacement]:
+    group_ids = message["group_ids"]
+    if not group_ids:
+        return None
+
+    query_time_flags = (EXCLUDE_GROUPS, message["project_id"], group_ids)
+    # XXX: Fugly, empty query
+    return Replacement("select 1", "select 1", {}, query_time_flags)
```
[the remark this answers is not in the document]
Comment (fpacifici) [reply snuba/datasets/errors_replacer.py:307]: I am not sure which use case will need to exclude groups. Sorry last time we discussed this was a long time ago and I am not sure I have context on the decision.

File: snuba/datasets/errors_replacer.py [hunk 2]
```diff
@@ -170,6 +170,10 @@ def process_message(self, message: ReplacementMessage) -> Optional[Replacement]:
                 self.__use_promoted_prewhere,
                 self.__schema,
             )
+        elif type_ == "tombstone_events":
+            processed = process_tombstone_events(event, self.__required_columns)
```
[the remark this answers is not in the document]
Comment (untitaker) [reply snuba/datasets/errors_replacer.py:174]: Unless the group is excluded this "event deletion" does not do anything immediately though, yeah.

File: tests/datasets/test_errors_replacer.py [hunk 3]
```diff
@@ -57,6 +59,21 @@ def _issue_count(self, project_id, group_id=None):
             self.app.post("/events_migration/query", data=json.dumps(args)).data
         )["data"]
 
+    def _get_group_id(self, project_id, event_id):
```
[the remark this answers is not in the document]
Comment (untitaker) [reply tests/datasets/test_errors_replacer.py]: yup

File: snuba/datasets/errors_replacer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (untitaker) [reply snuba/datasets/errors_replacer.py]: made the query attrs nullable

File: snuba/datasets/errors_replacer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkaemming) [reply snuba/datasets/errors_replacer.py:174]: Good point, I guess that trying to make this broadly generic is going to be confusing at some level so this is fine.

**Review by tkaemming — APPROVED** [review]

**Pull request merged**